### ISSUE TITLE
Some problems while building dogecoin on MAC OSX

### ISSUE THREAD

**Issue by bubudod** [dogecoin/dogecoin#1748]
After I installed berkeley-db of version 18.1.32 with homebrew, I ran

```shell
 ./configure --without-gui --with-incompatible-bdb
 make
```

However, some errors appeared in the output.

```log
Making all in src
  CXX      libdogecoin_server_a-validation.o
validation.cpp:182:111: error: use of undeclared identifier '_1'
        pool.NotifyEntryRemoved.connect(boost::bind(&MemPoolConflictRemovalTracker::NotifyEntryRemoved, this, _1, _2));
                                                                                                              ^
validation.cpp:182:115: error: use of undeclared identifier '_2'
        pool.NotifyEntryRemoved.connect(boost::bind(&MemPoolConflictRemovalTracker::NotifyEntryRemoved, this, _1, _2));
                                                                                                                  ^
validation.cpp:192:114: error: use of undeclared identifier '_1'
        pool.NotifyEntryRemoved.disconnect(boost::bind(&MemPoolConflictRemovalTracker::NotifyEntryRemoved, this, _1, _2));
                                                                                                                 ^
validation.cpp:192:118: error: use of undeclared identifier '_2'
        pool.NotifyEntryRemoved.disconnect(boost::bind(&MemPoolConflictRemovalTracker::NotifyEntryRemoved, this, _1, _2));
                                                                                                                     ^
In file included from validation.cpp:31:
In file included from ./txdb.h:10:
In file included from ./dbwrapper.h:11:
In file included from ./util.h:29:
In file included from /usr/local/include/boost/signals2/signal.hpp:22:
In file included from /usr/local/include/boost/signals2/connection.hpp:25:
In file included from /usr/local/include/boost/signals2/slot.hpp:15:
/usr/local/include/boost/bind/bind.hpp:376:9: error: no matching function for call to object of type 'boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>'
        unwrapper<F>::unwrap(f, 0)(a[base_type::a1_], a[base_type::a2_], a[base_type::a3_]);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/bind/bind.hpp:1296:16: note: in instantiation of function template specialization 'boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> >
      >::operator()<boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>, boost::_bi::rrlist2<std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason> >' requested here
        return l_( type<result_type>(), f_, a, 0 );
               ^
/usr/local/include/boost/function/function_template.hpp:158:33: note: in instantiation of function template specialization 'boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>,
      MemPoolRemovalReason>, boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > >::operator()<std::__1::shared_ptr<const CTransaction>,
      MemPoolRemovalReason>' requested here
          BOOST_FUNCTION_RETURN((*f)(BOOST_FUNCTION_ARGS));
                                ^
/usr/local/include/boost/function/function_template.hpp:940:53: note: in instantiation of member function 'boost::detail::function::void_function_obj_invoker2<boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker,
      std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>, boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > >, void, std::__1::shared_ptr<const
      CTransaction>, MemPoolRemovalReason>::invoke' requested here
        { { &manager_type::manage }, &invoker_type::invoke };
                                                    ^
/usr/local/include/boost/function/function_template.hpp:720:13: note: in instantiation of function template specialization 'boost::function2<void, std::__1::shared_ptr<const CTransaction>,
      MemPoolRemovalReason>::assign_to<boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>,
      boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > > >' requested here
      this->assign_to(f);
            ^
/usr/local/include/boost/function/function_template.hpp:1086:5: note: in instantiation of function template specialization 'boost::function2<void, std::__1::shared_ptr<const CTransaction>,
      MemPoolRemovalReason>::function2<boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>,
      boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > > >' requested here
    base_type(f)
    ^
/usr/local/include/boost/function/function_template.hpp:1139:5: note: in instantiation of function template specialization 'boost::function<void (std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason)>::function<boost::_bi::bind_t<void,
      boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>, boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>,
      boost::_bi::value<std::__1::placeholders::__ph<2> > > > >' requested here
    self_type(f).swap(*this);
    ^
/usr/local/include/boost/signals2/detail/slot_template.hpp:160:24: note: in instantiation of function template specialization 'boost::function<void (std::__1::shared_ptr<const CTransaction>,
      MemPoolRemovalReason)>::operator=<boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>,
      boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > > >' requested here
        _slot_function = detail::get_invocable_slot(f, detail::tag_type(f));
                       ^
/usr/local/include/boost/signals2/detail/slot_template.hpp:85:9: note: in instantiation of function template specialization 'boost::signals2::slot<void (std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason), boost::function<void
      (std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason)> >::init_slot_function<boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>,
      boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > > >' requested here
        init_slot_function(f);
        ^
validation.cpp:182:41: note: in instantiation of function template specialization 'boost::signals2::slot<void (std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason), boost::function<void
      (std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason)> >::slot<boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>,
      boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > > >' requested here
        pool.NotifyEntryRemoved.connect(boost::bind(&MemPoolConflictRemovalTracker::NotifyEntryRemoved, this, _1, _2));
                                        ^
/usr/local/include/boost/bind/mem_fn_template.hpp:278:7: note: candidate function not viable: no known conversion from 'std::__1::placeholders::__ph<2>' to 'MemPoolRemovalReason' for 3rd argument
    R operator()(T * p, A1 a1, A2 a2) const
      ^
/usr/local/include/boost/bind/mem_fn_template.hpp:283:25: note: candidate function template not viable: no known conversion from 'std::__1::placeholders::__ph<2>' to 'MemPoolRemovalReason' for 3rd argument
    template<class U> R operator()(U & u, A1 a1, A2 a2) const
                        ^
/usr/local/include/boost/bind/mem_fn_template.hpp:291:25: note: candidate function template not viable: no known conversion from 'std::__1::placeholders::__ph<2>' to 'MemPoolRemovalReason' for 3rd argument
    template<class U> R operator()(U const & u, A1 a1, A2 a2) const
                        ^
/usr/local/include/boost/bind/mem_fn_template.hpp:299:7: note: candidate function not viable: no known conversion from 'MemPoolConflictRemovalTracker *' to 'MemPoolConflictRemovalTracker &' for 1st argument; dereference the argument with *
    R operator()(T & t, A1 a1, A2 a2) const
      ^
In file included from validation.cpp:31:
In file included from ./txdb.h:10:
In file included from ./dbwrapper.h:11:
In file included from ./util.h:29:
In file included from /usr/local/include/boost/signals2/signal.hpp:22:
In file included from /usr/local/include/boost/signals2/connection.hpp:25:
In file included from /usr/local/include/boost/signals2/slot.hpp:15:
/usr/local/include/boost/bind/bind.hpp:107:19: error: invalid operands to binary expression ('const std::__1::placeholders::__ph<2>' and 'const std::__1::placeholders::__ph<2>')
        return t_ == rhs.t_;
               ~~ ^  ~~~~~~
/usr/local/include/boost/bind/bind.hpp:67:14: note: in instantiation of member function 'boost::_bi::value<std::__1::placeholders::__ph<2> >::operator==' requested here
    return a == b;
             ^
/usr/local/include/boost/bind/bind.hpp:395:13: note: in instantiation of function template specialization 'boost::_bi::ref_compare<boost::_bi::value<std::__1::placeholders::__ph<2> > >' requested here
            ref_compare( base_type::a3_, rhs.a3_, 0 );
            ^
/usr/local/include/boost/bind/bind.hpp:1413:51: note: in instantiation of member function 'boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > >::operator=='
      requested here
        return ref_compare( f_, rhs.f_, 0 ) && l_ == rhs.l_;
                                                  ^
/usr/local/include/boost/bind/bind.hpp:1497:14: note: in instantiation of member function 'boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>,
      boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > >::compare' requested here
    return a.compare(b);
             ^
/usr/local/include/boost/function/function_base.hpp:780:14: note: in instantiation of function template specialization 'boost::_bi::function_equal<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const
      CTransaction>, MemPoolRemovalReason>, boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > >' requested here
      return function_equal(*fp, g);
             ^
/usr/local/include/boost/signals2/detail/signal_template.hpp:530:46: note: in instantiation of function template specialization 'boost::operator==<boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker,
      std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>, boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > > >' requested here
            if((*it)->slot().slot_function() == slot)
                                             ^
/usr/local/include/boost/signals2/detail/signal_template.hpp:222:11: note: in instantiation of function template specialization 'boost::signals2::detail::signal_impl<void (std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason),
      boost::signals2::optional_last_value<void>, int, std::__1::less<int>, boost::function<void (std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason)>, boost::function<void (const boost::signals2::connection &,
      std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason)>, boost::signals2::mutex>::do_disconnect<boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>,
      MemPoolRemovalReason>, boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > > >' requested here
          do_disconnect(slot, is_group());
          ^
/usr/local/include/boost/signals2/detail/signal_template.hpp:718:19: note: in instantiation of function template specialization 'boost::signals2::detail::signal_impl<void (std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason),
      boost::signals2::optional_last_value<void>, int, std::__1::less<int>, boost::function<void (std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason)>, boost::function<void (const boost::signals2::connection &,
      std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason)>, boost::signals2::mutex>::disconnect<boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>,
      MemPoolRemovalReason>, boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > > >' requested here
        (*_pimpl).disconnect(slot);
                  ^
validation.cpp:192:33: note: in instantiation of function template specialization 'boost::signals2::signal<void (std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason), boost::signals2::optional_last_value<void>, int,
      std::__1::less<int>, boost::function<void (std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason)>, boost::function<void (const boost::signals2::connection &, std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason)>,
      boost::signals2::mutex>::disconnect<boost::_bi::bind_t<void, boost::_mfi::mf2<void, MemPoolConflictRemovalTracker, std::__1::shared_ptr<const CTransaction>, MemPoolRemovalReason>,
      boost::_bi::list3<boost::_bi::value<MemPoolConflictRemovalTracker *>, boost::arg<1>, boost::_bi::value<std::__1::placeholders::__ph<2> > > > >' requested here
        pool.NotifyEntryRemoved.disconnect(boost::bind(&MemPoolConflictRemovalTracker::NotifyEntryRemoved, this, _1, _2));
                                ^
/usr/local/include/boost/function/function_base.hpp:777:3: note: candidate function template not viable: no known conversion from 'const std::__1::placeholders::__ph<2>' to 'const boost::function_base' for 1st argument
  operator==(const function_base& f, Functor g)
  ^
/usr/local/include/boost/function/function_base.hpp:786:3: note: candidate function template not viable: no known conversion from 'const std::__1::placeholders::__ph<2>' to 'const boost::function_base' for 2nd argument
  operator==(Functor g, const function_base& f)
  ^
/usr/local/include/boost/blank.hpp:58:13: note: candidate function not viable: no known conversion from 'const std::__1::placeholders::__ph<2>' to 'const boost::blank' for 1st argument
inline bool operator==(const blank&, const blank&)
            ^
/usr/local/include/boost/function/function_base.hpp:715:13: note: candidate function not viable: no known conversion from 'const std::__1::placeholders::__ph<2>' to 'const boost::function_base' for 1st argument
inline bool operator==(const function_base& f,
            ^
/usr/local/include/boost/function/function_base.hpp:727:13: note: candidate function not viable: no known conversion from 'const std::__1::placeholders::__ph<2>' to 'detail::function::useless_clear_type *' for 1st argument
inline bool operator==(detail::function::useless_clear_type*,
            ^
/usr/local/include/boost/smart_ptr/shared_ptr.hpp:832:40: note: candidate template ignored: could not match 'shared_ptr' against '__ph'
template<class T, class U> inline bool operator==(shared_ptr<T> const & a, shared_ptr<U> const & b) BOOST_SP_NOEXCEPT
                                       ^
/usr/local/include/boost/smart_ptr/shared_ptr.hpp:855:31: note: candidate template ignored: could not match 'shared_ptr' against '__ph'
template<class T> inline bool operator==( shared_ptr<T> const & p, boost::detail::sp_nullptr_t ) BOOST_SP_NOEXCEPT
                              ^
/usr/local/include/boost/smart_ptr/shared_ptr.hpp:860:31: note: candidate template ignored: could not match 'shared_ptr' against '__ph'
template<class T> inline bool operator==( boost::detail::sp_nullptr_t, shared_ptr<T> const & p ) BOOST_SP_NOEXCEPT
                              ^
/usr/local/include/boost/unordered/unordered_set.hpp:1497:17: note: candidate template ignored: could not match 'unordered_set' against '__ph'
    inline bool operator==(
                ^
/usr/local/include/boost/unordered/unordered_set.hpp:1900:17: note: candidate template ignored: could not match 'unordered_multiset' against '__ph'
    inline bool operator==(unordered_multiset<T, H, P, A> const& m1,
                ^
/usr/local/include/boost/unordered/unordered_map_fwd.hpp:27:17: note: candidate template ignored: could not match 'unordered_map' against '__ph'
    inline bool operator==(
                ^
/usr/local/include/boost/unordered/unordered_map_fwd.hpp:43:17: note: candidate template ignored: could not match 'unordered_multimap' against '__ph'
    inline bool operator==(unordered_multimap<K, T, H, P, A> const&,
                ^
/usr/local/include/boost/function/function_base.hpp:814:3: note: candidate template ignored: could not match 'reference_wrapper' against '__ph'
  operator==(const function_base& f, reference_wrapper<Functor> g)
  ^
/usr/local/include/boost/function/function_base.hpp:823:3: note: candidate template ignored: could not match 'reference_wrapper' against '__ph'
  operator==(reference_wrapper<Functor> g, const function_base& f)
  ^
/usr/local/include/boost/function/function_template.hpp:1042:8: note: candidate template ignored: could not match 'function0' against '__ph'
  void operator==(const BOOST_FUNCTION_FUNCTION<
       ^
/usr/local/include/boost/function/function_template.hpp:1042:8: note: candidate template ignored: could not match 'function1' against '__ph'
  void operator==(const BOOST_FUNCTION_FUNCTION<
       ^
/usr/local/include/boost/function/function_template.hpp:1042:8: note: candidate template ignored: could not match 'function2' against '__ph'
  void operator==(const BOOST_FUNCTION_FUNCTION<
       ^
/usr/local/include/boost/function/function_template.hpp:1042:8: note: candidate template ignored: could not match 'function3' against '__ph'
  void operator==(const BOOST_FUNCTION_FUNCTION<
       ^
/usr/local/include/boost/function/function_template.hpp:1042:8: note: candidate template ignored: could not match 'function4' against '__ph'
  void operator==(const BOOST_FUNCTION_FUNCTION<
       ^
/usr/local/include/boost/function/function_template.hpp:1042:8: note: candidate template ignored: could not match 'function5' against '__ph'
  void operator==(const BOOST_FUNCTION_FUNCTION<
       ^
/usr/local/include/boost/function/function_template.hpp:1042:8: note: candidate template ignored: could not match 'function6' against '__ph'
  void operator==(const BOOST_FUNCTION_FUNCTION<
       ^
/usr/local/include/boost/function/function_template.hpp:1042:8: note: candidate template ignored: could not match 'function7' against '__ph'
  void operator==(const BOOST_FUNCTION_FUNCTION<
       ^
/usr/local/include/boost/function/function_template.hpp:1042:8: note: candidate template ignored: could not match 'function8' against '__ph'
  void operator==(const BOOST_FUNCTION_FUNCTION<
       ^
/usr/local/include/boost/function/function_template.hpp:1042:8: note: candidate template ignored: could not match 'function9' against '__ph'
  void operator==(const BOOST_FUNCTION_FUNCTION<
       ^
/usr/local/include/boost/function/function_template.hpp:1042:8: note: candidate template ignored: could not match 'function10' against '__ph'
  void operator==(const BOOST_FUNCTION_FUNCTION<
       ^
/usr/local/include/boost/bind/arg.hpp:48:40: note: candidate template ignored: could not match 'arg' against '__ph'
template< int I > BOOST_CONSTEXPR bool operator==( arg<I> const &, arg<I> const & )
                                       ^
6 errors generated.
make[2]: *** [libdogecoin_server_a-validation.o] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all-recursive] Error 1
```

Then I tried to reinstall the berkeley db of version 5.1.29, but it still can not solve problems. the log is 

```log
checking for visibility attribute... yes
checking for Berkeley DB C++ headers... no
configure: error: libdb_cxx headers missing, Dogecoin Core requires this library for wallet functionality (--disable-wallet to disable wallet functionality)
./configure: line 357: sort: No such file or directory
./configure: line 354: sed: No such file or directory
```

Finally, when building berkeley db, I tried to add `--enable-cxx`.

```shell
../dist/configure --prefix=/usr/local/Cellar/berkeleydb/5_1_29 --enable-cxx
make
```

lots of errors came out too.

```log
./libtool --mode=compile g++ -c -I. -I../src -I/usr/local/opt/icu4c/include -O ../lang/cxx/cxx_db.cpp
libtool: compile:  g++ -c -I. -I../src -I/usr/local/opt/icu4c/include -O ../lang/cxx/cxx_db.cpp  -fno-common -DPIC -o .libs/cxx_db.o
In file included from ../lang/cxx/cxx_db.cpp:13:
In file included from ./db_cxx.h:55:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iostream:37:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/ios:215:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__locale:14:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string:504:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string_view:175:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__string:57:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:643:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:666:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:1742:13: error: expected unqualified-id
atomic_init(volatile atomic<_Tp>* __o, _Tp __d) _NOEXCEPT
            ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:1742:13: error: expected ')'
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:1742:1: note: to match this '('
atomic_init(volatile atomic<_Tp>* __o, _Tp __d) _NOEXCEPT
^
../src/dbinc/atomic.h:73:30: note: expanded from macro 'atomic_init'
#define atomic_init(p, val)     ((p)->value = (val))
                                 ^
In file included from ../lang/cxx/cxx_db.cpp:13:
In file included from ./db_cxx.h:55:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iostream:37:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/ios:215:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__locale:14:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string:504:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string_view:175:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__string:57:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:643:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:666:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:1742:1: error: expected ')'
atomic_init(volatile atomic<_Tp>* __o, _Tp __d) _NOEXCEPT
^
../src/dbinc/atomic.h:73:33: note: expanded from macro 'atomic_init'
#define atomic_init(p, val)     ((p)->value = (val))
                                    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:1742:1: note: to match this '('
../src/dbinc/atomic.h:73:29: note: expanded from macro 'atomic_init'
#define atomic_init(p, val)     ((p)->value = (val))
                                ^
In file included from ../lang/cxx/cxx_db.cpp:13:
In file included from ./db_cxx.h:55:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iostream:37:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/ios:215:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__locale:14:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string:504:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string_view:175:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__string:57:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:643:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:666:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:2315:35: error: unknown type name 'atomic_flag'
atomic_flag_test_and_set(volatile atomic_flag* __o) _NOEXCEPT
                                  ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:2322:26: error: unknown type name 'atomic_flag'
atomic_flag_test_and_set(atomic_flag* __o) _NOEXCEPT
                         ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:2329:44: error: unknown type name 'atomic_flag'
atomic_flag_test_and_set_explicit(volatile atomic_flag* __o, memory_order __m) _NOEXCEPT
                                           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:2336:35: error: unknown type name 'atomic_flag'
atomic_flag_test_and_set_explicit(atomic_flag* __o, memory_order __m) _NOEXCEPT
                                  ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:2343:28: error: unknown type name 'atomic_flag'
atomic_flag_clear(volatile atomic_flag* __o) _NOEXCEPT
                           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:2350:19: error: unknown type name 'atomic_flag'
atomic_flag_clear(atomic_flag* __o) _NOEXCEPT
                  ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:2357:37: error: unknown type name 'atomic_flag'
atomic_flag_clear_explicit(volatile atomic_flag* __o, memory_order __m) _NOEXCEPT
                                    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:2364:28: error: unknown type name 'atomic_flag'
atomic_flag_clear_explicit(atomic_flag* __o, memory_order __m) _NOEXCEPT
                           ^
11 errors generated.
make: *** [cxx_db.lo] Error 1
```

Because of those problems. I can not build the dogecoin on my laptop. So, I hope you could give me a help.

**Comment by esantonja** [comment]
Hi @bubudod.
I was facing the same issues. 

I've created a brew formula and added to this repo 'https://github.com/esantonja/homebrew-formulas' in case anybody else has the same problem.

The next problem I faced was openssl, review that you got the latest version installed with brew, add the openssl into the PATH of your system and create a sym link to the include folder. As explained on this thread https://stackoverflow.com/questions/35129977/how-to-install-latest-version-of-openssl-mac-os-x-el-capitan

Hope this helps.
